.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1732,6 +1734,49 @@ private DateTimeResolutionResult ParseDuration(string text, DateObject reference
                         endDate = modAndDateResult.EndDate;
                     }
 
+                    // Handle cases like "first 2 weeks of 2019", "last 3 months of this year"
+                    var matchBefore = this.config.FirstLastRegex.Match(beforeStr);
```

Comment:
Is there any kind of "pre-check" that can happen here to avoid applying both regexes always to parsing? Or is this already nested somewhere that isn't called frequently?

Reply:
now the 2 checks are nested